### DISCUSSION TITLE
IOS 2.16.30 Beta

### DISCUSSION THREAD

**Discussion by bco116** [cagnulein/qdomyos-zwift#2000] [General]
Hello! I recently downloaded QZ to interface my Nordictrack 1750 treadmill with Zwift, and it works great. Now I am reading where automatic inclination adjustment is now possible using QZ 2.16.30 - is it possible to receive a TestFlight code for this version?

**Comment by cagnulein** [comment]
hi! are you able to change inclination from the qz app directly? that's the
first requirement for auto inclination

let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 14 gen 2024 alle 20:21 bco116 ***@***.***> ha
scritto:

**Reply by bco116** [reply]
Hello, yes I am.  I needed to play with the offset; sometimes when I pressed +, the incline would keep going down.  Once I adjusted this setting through some trial and error, + and - appear to be working correctly.  Thanks for your help, the potential of this app is amazing!

**Comment by cagnulein** [comment]
Glad to hear it!
which offset are you referring?
anyway send me an email to roberto.viola83 at gmail.com and I will send you
the beta
mention this ticket in the email


Il giorno dom 14 gen 2024 alle 21:42 bco116 ***@***.***> ha
scritto:

**Comment by bco116** [comment]
The offset is “inclination override offset”, under treadmill inclination overrides.  Maybe I did not need to adjust this, but it seemed to help. Email coming!On Jan 14, 2024, at 3:45 PM, Roberto Viola ***@***.***> wrote:﻿
Glad to hear it!

which offset are you referring?

anyway send me an email to roberto.viola83 at gmail.com and I will send you

the beta

mention this ticket in the email





Il giorno dom 14 gen 2024 alle 21:42 bco116 ***@***.***> ha

scritto:

**Comment by cagnulein** [comment]
hah ok so it means that the coordinates are slightly different from the
implementation

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 14 gen 2024 alle 21:58 bco116 ***@***.***> ha
scritto:

**Comment by bco116** [comment]
Yep, that’s what I figured.  I am running the old version of iFit Cardio that came with my treadmill, because if I let it do the software update, it would no longer communicate with QZ.Sent from my iPadOn Jan 14, 2024, at 4:01 PM, Roberto Viola ***@***.***> wrote:﻿
hah ok so it means that the coordinates are slightly different from the

implementation



Roberto Viola

Software engineer and open source enthusiast

http://robertoviola.cloud







Il giorno dom 14 gen 2024 alle 21:58 bco116 ***@***.***> ha

scritto:

**Comment by cagnulein** [comment]
ok in case I can add coordinate for this
try the beta first :)

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 14 gen 2024 alle 22:05 bco116 ***@***.***> ha
scritto:

**Comment by bco116** [comment]
The beta works! I chose a hilly route in Zwift, and inclination adjusted automatically as I moved through the route. It appeared to follow more closely on the negative side than on the positive.  For example, if Zwift incline was 5%, the treadmill would be at 2.5 or 3.  On the downhills, it was usually just about spot on.  With auto inclination, is QZ able to directly set the treadmill inclination value, or is it “touching” the sliders like it does for manual +/- adjustments?  Just wondering if I need to adjust my offsets some more.  Thanks again for your help!On Jan 14, 2024, at 4:18 PM, Roberto Viola ***@***.***> wrote:﻿
ok in case I can add coordinate for this

try the beta first :)



Roberto Viola

Software engineer and open source enthusiast

http://robertoviola.cloud







Il giorno dom 14 gen 2024 alle 22:05 bco116 ***@***.***> ha

scritto:

**Comment by cagnulein** [comment]
that's great! qz is controlling always the inclination using the virtual
touch events, so you can use the inclination offset setting to fix it or we
can try to handle the coordinates of your ifit app

the first should be easier for you since you can do by yourself :)

let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 14 gen 2024 alle 22:34 bco116 ***@***.***> ha
scritto:

**Comment by bco116** [comment]
Hi Roberto - I’m wondering if you might be able to assist with the inclination offset settings. I played around with them yesterday on QZ 2.16.30, but I could only make the manual inclination adjustment work in the negative direction, no matter what value I selected for the offset. However, when using with Zwift, it appeared to automatically adjust in both the positive and negative directions. Unfortunately, I inadvertently picked a mostly downhill route, so it wasn’t a good test - I’ll try again tonight on a more varied course.I have 2 questions related to the inclination offset:1) what is the unit of measurement of the value being set? Is it pixels? Percent incline?2) when I adjust the offset, do I also need to set the individual override for each half percent? For example, if I set the offset to 5, do I need to override 0% to 5, 0.5% to 5.5, and so on?Thanks for your help!On Jan 14, 2024, at 4:37 PM, Roberto Viola ***@***.***> wrote:﻿
that's great! qz is controlling always the inclination using the virtual

touch events, so you can use the inclination offset setting to fix it or we

can try to handle the coordinates of your ifit app



the first should be easier for you since you can do by yourself :)



let me know



Roberto Viola

Software engineer and open source enthusiast

http://robertoviola.cloud







Il giorno dom 14 gen 2024 alle 22:34 bco116 ***@***.***> ha

scritto:

**Reply by cagnulein** [reply]
it's in % grade. no you don't need to change the override

i suggest to map correctly your touchscreen

**Comment by bco116** [comment]
Is that something you’d be able to assist with if I provide the iFit version?On Jan 16, 2024, at 11:21 PM, Roberto Viola ***@***.***> wrote:﻿
it's in % grade. no you don't need to change the override
i suggest to map correctly your touchscreen

**Reply by cagnulein** [reply]
in order to do so you have to use the https://github.com/cagnulein/QZCompanionNordictrackTreadmill/blob/master/InstallPackage/adb-screenshot.bat contained here https://github.com/cagnulein/QZCompanionNordictrackTreadmill/archive/refs/tags/3.3.27.zip

let me know if you need help

**Reply by bco116** [reply]
Hi Roberto - I was able to successfully generate a screenshot, please see below:

![Sat01-20-2024@16-27-45 20](https://github.com/cagnulein/qdomyos-zwift/assets/156609223/91764ab8-6f3a-45b3-a464-81a612caae6d)

**Reply by cagnulein** [reply]
ok thanks, i will check this in the next days!

**Reply by cagnulein** [reply]
@bco116 could you please tell me which is your firmware version of the treadmill and the specific treadmill version (share me a screenshot from the android settings of the tablet). So I will add a new settings in the qz companion with this exact name. Let me know!

**Reply by bco116** [reply]
I wasn't able to get adb-screenshot.bat to work tonight for some reason.  The treadmill is a 2020 NordicTrack C1750, running iFit Cardio version 2.2.9.378.  Here is a photo of the settings from the app:

![IMG_3577](https://github.com/cagnulein/qdomyos-zwift/assets/156609223/648cf8b0-1033-4a19-ac74-aaaf18067c38)

**Reply by cagnulein** [reply]
new apk ready @bco116 https://github.com/cagnulein/QZCompanionNordictrackTreadmill/releases/download/3.3.30/QZCompanionNordictrackTreadmill.apk

set the new c1750 2020 setting and restart the treadmill
let me know if the coordinates are good now

**Comment by bco116** [comment]
Ok thanks, I’ll try it tonight!On Jan 17, 2024, at 11:50 AM, Roberto Viola ***@***.***> wrote:﻿
in order to do so you have to use the https://github.com/cagnulein/QZCompanionNordictrackTreadmill/blob/master/InstallPackage/adb-screenshot.bat contained here https://github.com/cagnulein/QZCompanionNordictrackTreadmill/archive/refs/tags/3.3.27.zip
let me know if you need help

**Comment by bco116** [comment]
I tried this last night, but I think the mapping is still a bit off. The manual incline +/- buttons in the QZ app would not respond unless I set the incline on the treadmill first. Once I did this, I could use the buttons in the app, but pressing the decrease button would still make the incline increase. Once the incline reached 4.5, the +/- buttons in the app no longer functionedOn Jan 24, 2024, at 3:47 AM, Roberto Viola ***@***.***> wrote:﻿
new apk ready @bco116 https://github.com/cagnulein/QZCompanionNordictrackTreadmill/releases/download/3.3.30/QZCompanionNordictrackTreadmill.apk
set the new c1750 2020 setting and restart the treadmill
let me know if the coordinates are good now

**Reply by cagnulein** [reply]
ok i guess you have to validate the formula that I added here via adb

```
x1 = 75;
y1Inclination = 520 - (int) (QZService.lastInclinationFloat * 20);
y2 = y1Inclination - (int) ((reqInclination - QZService.lastInclinationFloat) * 20);

adb shell input swipe x1 y1Inclination x1 y2 200
```

are you able to test it via adb?
